### PR TITLE
Add pipe cleanup docs

### PR DESCRIPTION
**Pull request by karenfeng**
## What changes are proposed in this pull request?

Adds documentation for pipe cleanup.

## How is this patch tested?
- [ ] Unit tests
- [ ] Integration tests
- [x] Manual tests

`make livehtml` looks good

### CODE REVIEW

File: docs/source/tertiary/pipe-transformer.rst [hunk 1]
```diff
@@ -128,4 +128,14 @@ The following options are not supported:
  - ``path`` options are ignored
  - The ``parserLib`` option is ignored. ``univocity`` is always used as the CSV parsing library.
 
+Cleanup
+=======
+
+The pipe transformer performs RDD caching to optimize performance. Spark automatically drops out old data partitions in
+a least-recently-used (LRU) fashion. If you would like to manually clean up the RDDs cached by the pipe transformer
+instead of waiting for them to fall out of the cache, use the pipe cleanup transformer on any DataFrame.
+
```
Comment (henrydavidge) [inline docs/source/tertiary/pipe-transformer.rst]: Note that this should be done after the result of the pipe transformer has been written out (or however it's ultimately materialized).

File: docs/source/tertiary/pipe-transformer.rst [hunk 2]
```diff
@@ -128,4 +128,14 @@ The following options are not supported:
  - ``path`` options are ignored
  - The ``parserLib`` option is ignored. ``univocity`` is always used as the CSV parsing library.
 
+Cleanup
+=======
+
+The pipe transformer performs RDD caching to optimize performance. Spark automatically drops out old data partitions in
```
Comment (henrydavidge) [inline docs/source/tertiary/pipe-transformer.rst]: nit: performs => uses

Comment (henrydavidge) [reply docs/source/tertiary/pipe-transformer.rst]: remove "out"

File: docs/source/tertiary/pipe-transformer.rst [hunk 3]
```diff
@@ -128,4 +128,15 @@ The following options are not supported:
  - ``path`` options are ignored
  - The ``parserLib`` option is ignored. ``univocity`` is always used as the CSV parsing library.
 
+Cleanup
+=======
+
+The pipe transformer uses RDD caching to optimize performance. Spark automatically drops old data partitions in a
+least-recently-used (LRU) fashion. If you would like to manually clean up the RDDs cached by the pipe transformer
+instead of waiting for them to fall out of the cache, use the pipe cleanup transformer on any DataFrame. Do not perform
+cleanup until the pipe transformer results have been materialized, such as by being written to Delta.
```
Comment (williambrandler) [inline docs/source/tertiary/pipe-transformer.rst]: Delta Tables (with hyperlink to Delta.io)

**Pull request merged**